python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -105,54 +107,56 @@ def register_tuple(
                 Can be None to gracefull load previously serialized objects as None.
             serializer: The class to use when serializing and deserializing
         """
-        args_for_class = signature(named_tuple_class.__new__).parameters
-        self.tuples[name] = (
-            named_tuple_class,
-            serializer or DefaultNamedTupleSerializer,
-            args_for_class,
+        serializer_class = serializer_class or NamedTupleSerializer
+        serializer = serializer_class(
+            klass=named_tuple_class,
+            storage_name=storage_name,
+            storage_field_names=storage_field_names,
+            old_fields=old_fields,
+            skip_when_empty_fields=skip_when_empty_fields,
```

Comment:
IIUC that would be a potential issue if serializer instances were created each time we wanted to serialize/deserialize an object. That's not what this PR does. It just replaces the classes that used to be in the registry with instances-- the same instances are used repeatedly, just like the same classes were used repeatedly. The reason for the replacement was just to make these reused entities easier to parameterize via instance attributes (e.g. before, `skip_empty_fields` was implemented as a class method).

Curious about `pympler.muppy`, I'll give it a shot.

Reply:
The concern is the contents of the global whitelist map that is populated at import time and retained for the life of the process acting as a fixed memory cost. Previously there were many entries pointing at the same default class, and now each entry has its own object instance. The whitelist gets pretty big, especially in internal, and i saw the `Parameter` entries that are in the map in the current impl during recent memory profiling so it was top of mind.

Reply:
ah I see-- well I tried this on both `master` and this branch `sean/refactor-serdes-2`):

```
from pympler import asizeof
from dagster.serdes._serdes import _WHITELIST_MAP
asizeof.asizeof(_WHITELIST_MAP)
```

- `master`: 115792
- `sean/refactor-serdes-2`: 76472

So the size appears substantially _reduced_ with this refactor-- though in either case the size is pretty small.

I'm not sure why the size is smaller-- perhaps the elimination of the `serialized_names` and `deserialized_names` tables has a bigger effect than storing object instances? I don't really know how much memory an object instance takes up if the instance attributes are mostly `None` (as they are in this case) but unless the above experiment is missing something it appears to be very little.

Reply:
Cool thanks for poking at it. Couple possible things 

* ensuring the whitelist map has been populated sufficiently via imports. Going through `dagster` *should* get most of them 

* im not certain of using `asizeof` is how the recursing and totaling works. One possible explanation is that more was counted as part of the previous it was a pile of basic data structures and in the newer one it only counts its references to serializer instances instead of the size of those as well

One other thing to cross reference would be looking at the summary for all objects and seeing how `WhitelistMap` and the serializer classes net out
https://pympler.readthedocs.io/en/latest/muppy.html#the-summary-module